packages/calcite-components/src/components/input/input.scss

```diff
@@ -257,6 +257,11 @@ input:focus {
 
 // ICONS
 
+// type
+:host([icon="check-square-f"]) {
```

Comment:
Can you apply `isolation: isolate` to the internal input's [wrapper element](https://github.com/Esri/calcite-design-system/blob/3111a343b71db9cf394aebc089cdef2009dbaa79/packages/calcite-components/src/components/input/input.tsx#L1155). Unlike the host, the wrapper doesn't support slotted children, so it won't conflict with user-land z-indices.